source/common/http/http2/codec_impl.cc

```diff
@@ -1847,9 +1852,11 @@ ConnectionImpl::Http2Callbacks::Http2Callbacks() {
       });
 
   nghttp2_session_callbacks_set_on_frame_not_send_callback(
-      callbacks_, [](nghttp2_session*, const nghttp2_frame*, int, void*) -> int {
+      callbacks_, [](nghttp2_session*, const nghttp2_frame* frame, int, void* user_data) -> int {
         // We used to always return failure here but it looks now this can get called if the other
         // side sends GOAWAY and we are trying to send a SETTINGS ACK. Just ignore this for now.
+        static_cast<ConnectionImpl*>(user_data)->onFrameNotSend(
```

Comment:
why NotSend vs NotSent?

Reply:
The nghttp2 callback is called `on_frame_not_send_callback`. I don't feel strongly about the name, though.